version: "3"

services:
  validator1:
    networks:
      pera-network:
        ipv4_address: 10.0.0.11
    image: pera-io/pera-node:mainnet
    container_name: validator1
    hostname: validator1
    environment:
      - RUST_BACKTRACE=1
      - RUST_LOG=info,pera_core=debug,pera_network=debug,pera_node=debug,narwhal=debug,narwhal-primary::helper=info,jsonrpsee=error
      - RPC_WORKER_THREAD=12
      - NEW_CHECKPOINT_WARNING_TIMEOUT_MS=30000
      - NEW_CHECKPOINT_PANIC_TIMEOUT_MS=60000
    volumes:
      - ./genesis/files/validator1-8080.yaml:/opt/pera/config/validator.yaml:ro
      - ./genesis/files/genesis.blob:/opt/pera/config/genesis.blob:ro
      - /tmp/pera/db1:/opt/pera/db:rw
    command:
      [
        "/usr/local/bin/pera-node",
        "--config-path",
        "/opt/pera/config/validator.yaml",
      ]
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-file: "10"
        max-size: "1g"
  validator2:
    networks:
      pera-network:
        ipv4_address: 10.0.0.12
    image: pera-io/pera-node:mainnet
    container_name: validator2
    hostname: validator2
    environment:
      - RUST_BACKTRACE=1
      - RUST_LOG=info,pera_core=debug,pera_network=debug,pera_node=debug,narwhal=debug,narwhal-primary::helper=info,jsonrpsee=error
      - RPC_WORKER_THREAD=12
      - NEW_CHECKPOINT_WARNING_TIMEOUT_MS=30000
      - NEW_CHECKPOINT_PANIC_TIMEOUT_MS=60000
    volumes:
      - ./genesis/files/validator2-8080.yaml:/opt/pera/config/validator.yaml:ro
      - ./genesis/files/genesis.blob:/opt/pera/config/genesis.blob:ro
      - /tmp/pera/db2:/opt/pera/db:rw
    command:
      [
        "/usr/local/bin/pera-node",
        "--config-path",
        "/opt/pera/config/validator.yaml",
      ]
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-file: "10"
        max-size: "1g"
  validator3:
    networks:
      pera-network:
        ipv4_address: 10.0.0.13
    image: pera-io/pera-node:mainnet
    container_name: validator3
    hostname: validator3
    environment:
      - RUST_BACKTRACE=1
      - RUST_LOG=info,pera_core=debug,pera_network=debug,pera_node=debug,narwhal=debug,narwhal-primary::helper=info,jsonrpsee=error
      - RPC_WORKER_THREAD=12
      - NEW_CHECKPOINT_WARNING_TIMEOUT_MS=30000
      - NEW_CHECKPOINT_PANIC_TIMEOUT_MS=60000
    volumes:
      - ./genesis/files/validator3-8080.yaml:/opt/pera/config/validator.yaml:ro
      - ./genesis/files/genesis.blob:/opt/pera/config/genesis.blob:ro
      - /tmp/pera/db3:/opt/pera/db:rw
    command:
      [
        "/usr/local/bin/pera-node",
        "--config-path",
        "/opt/pera/config/validator.yaml",
      ]
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-file: "10"
        max-size: "1g"
  validator4:
    networks:
      pera-network:
        ipv4_address: 10.0.0.14
    image: pera-io/pera-node:mainnet
    container_name: validator4
    hostname: validator4
    environment:
      - RUST_BACKTRACE=1
      - RUST_LOG=info,pera_core=debug,pera_network=debug,pera_node=debug,narwhal=debug,narwhal-primary::helper=info,jsonrpsee=error
      - RPC_WORKER_THREAD=12
      - NEW_CHECKPOINT_WARNING_TIMEOUT_MS=30000
      - NEW_CHECKPOINT_PANIC_TIMEOUT_MS=60000
    volumes:
      - ./genesis/files/validator4-8080.yaml:/opt/pera/config/validator.yaml:ro
      - ./genesis/files/genesis.blob:/opt/pera/config/genesis.blob:ro
      - /tmp/pera/db4:/opt/pera/db:rw
    command:
      [
        "/usr/local/bin/pera-node",
        "--config-path",
        "/opt/pera/config/validator.yaml",
      ]
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-file: "10"
        max-size: "1g"
  fullnode1:
    networks:
      pera-network:
        ipv4_address: 10.0.0.15
    image: pera-io/pera-node:mainnet
    hostname: fullnode1
    container_name: fullnode1
    environment:
      - RUST_BACKTRACE=1
      - RUST_LOG=info,pera_core=debug,pera_network=debug,pera_node=debug,narwhal=debug,narwhal-primary::helper=info,jsonrpsee=error
      - RPC_WORKER_THREAD=12
      - NEW_CHECKPOINT_WARNING_TIMEOUT_MS=30000
      - NEW_CHECKPOINT_PANIC_TIMEOUT_MS=60000
    volumes:
      - ./genesis/static/fullnode.yaml:/opt/pera/config/fullnode.yaml:ro
      - ./genesis/files/genesis.blob:/opt/pera/config/genesis.blob:ro
      - /tmp/pera/db5:/opt/pera/db:rw
    command:
      [
        "/usr/local/bin/pera-node",
        "--config-path",
        "/opt/pera/config/fullnode.yaml",
      ]
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-file: "10"
        max-size: "1g"

  stress:
    networks:
      pera-network:
        ipv4_address: 10.0.0.16
    image: stress:testing
    container_name: stress
    environment:
      - RUST_LOG=info
      - STARTUP_DELAY_SECONDS=5
      - STRESS_STAGGERED_START_MAX_MULTIPLIER=0
      - FULLNODE_RPC_ADDRESS=10.0.0.15:9000
      - USE_FULLNODE_FOR_RECONFIG=false
      - PRIMARY_GAS_OWNER=0xd59d79516a4ed5b6825e80826c075a12bdd2759aaeb901df2f427f5f880c8f60
      - GENESIS_BLOB_PATH=/opt/pera/config/genesis.blob
      - KEYSTORE_PATH=/opt/pera/config/pera.keystore
      - STRESS_TARGET_QPS=10
      - STRESS_SHARED_COUNTER=1
      - STRESS_TRANSFER_OBJECT=1
      - STRESS_DELEGATION=0
      - BATCH_PAYMENT=1
      - BATCH_PAYMENT_SIZE=100
      - STRESS_ADVERSARIAL=0
    volumes:
      - ./genesis/files/genesis.blob:/opt/pera/config/genesis.blob:ro
      - ./genesis/static/pera.keystore:/opt/pera/config/pera.keystore:ro
    restart: on-failure
    logging:
      driver: "json-file"
      options:
        max-file: "10"
        max-size: "1g"
networks:
  pera-network:
    driver: bridge
    ipam:
      config:
      - subnet: 10.0.0.0/24
